h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -341,7 +341,7 @@ protected void validateRowSampleRate() {
         final int [] cons = new int[_nclass];
         for( int i=0; i<_nclass; i++ ) {
           names[i] = "NIDs_" + domain[i];
-          cons[i] = isSupervised() && ((_parms._distribution == DistributionFamily.quasibinomial || _model._output._distribution[i]==0)) ? -1 : 0;
```

Comment:
This code causes the first tree in Quasibinomial GBM is null even though the target distribution is calculated correctly for {0, 1} domains. For other domains like {0, 42}, the target distribution calculation throws an exception. See other changes bellow.